### DISCUSSION TITLE
Progress Callback for Downloading Models

### DISCUSSION THREAD

**Discussion by fakerybakery** [huggingface/diffusers#9437] [Q&A] [answered]
Hi,
Is there any way to capture the progress while a pipeline is downloading a model?
Thanks!

**Comment by bmaxdk** [comment]
Yes, you can use `huggingface_hub.snapshot_download` to capture the progress while a pipeline is downloading a model. It automatically shows a progress bar using tqdm
here is a [link](https://huggingface.co/docs/huggingface_hub/en/package_reference/file_download#huggingface_hub.snapshot_download)

**Reply by fakerybakery** [reply]
Thanks!
Is there a way to use a custom callback for downloads?

**Reply by bmaxdk** [reply, marked as answer]
Yes, you can use your own custom calllback. in that case, you can use [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download).

```py
hf_hub_download(repo_id=repo_id, filename=filename, progress_callback=progress_callback)
```

hf_hub_download suposrt `hf_hub_download` function does support a your own progress_callback parameter to track download progress

**Reply by fakerybakery** [reply]
Thanks!

**Reply by phlmn** [reply]
> Yes, you can use your own custom calllback. in that case, you can use [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download).
> 
> ```python
> hf_hub_download(repo_id=repo_id, filename=filename, progress_callback=progress_callback)
> ```
> 
> hf_hub_download suposrt `hf_hub_download` function does support a your own progress_callback parameter to track download progress

Has this been removed? In version 0.28.1 the parameter does not exist.

**Reply by Josh-Weston** [reply]
Parameter does not exist in 0.33.4.
